R/Belgium.R

```diff
@@ -0,0 +1,208 @@
+        tally(CASES) %>%
+        ungroup()
+
+      hosp_data <- self$data$raw$hosp %>%
+        select(DATE, REGION, NEW_IN) %>%
+        mutate(
+          DATE = ymd(DATE),
+          NEW_IN = as.numeric(NEW_IN)
+        ) %>%
+        replace_na(list(REGION = "Unknown")) %>%
+        group_by(DATE, REGION) %>%
+        tally(wt = NEW_IN) %>%
+        ungroup()
+
+      deaths_data <- self$data$raw$deaths %>%
+        select(DATE, REGION, DEATHS) %>%
+        mutate(
+          DATE = ymd(DATE),
+          DEATHS = as.numeric(DEATHS)
+        ) %>%
+        replace_na(list(REGION = "Unknown")) %>%
+        group_by(DATE, REGION) %>%
+        tally(wt = DEATHS) %>%
+        ungroup()
+
+      # Join the three datasets and rename columns
+      cases_and_hosp_data <- full_join(cases_data,
+                                       hosp_data,
+                                       by = c("DATE", "REGION"))
+
+      all_data <- full_join(cases_and_hosp_data,
+                                   deaths_data,
+                                   by = c("DATE", "REGION")) %>%
+        rename(date = DATE, level_1_region = REGION,
+               cases_new = n.x, hosp_new = n.y, deaths_new = n)
+      self$data$clean <-
+        left_join(all_data, self$codes_lookup[[1]],
+                  by = c("level_1_region"),
+                  copy = TRUE)
+    },
+
+    #' @description Belgium Specific Province Level Data Cleaning
+    # nolint start
+    #' @importFrom dplyr group_by summarise ungroup full_join tally mutate select rename
+    #' @importFrom tidyr replace_na
+    #' @importFrom lubridate ymd
+    # nolint end
+    #'
+    clean_level_2 = function() {
+      cases_data <- self$data$raw$main %>%
+        select(DATE, REGION, PROVINCE, CASES) %>%
+        mutate(
+          DATE = ymd(DATE),
+          CASES = as.numeric(CASES)
+        ) %>%
+        replace_na(list(
+          REGION = "Unknown",
+          PROVINCE = "Unknown"
+        )) %>%
+        group_by(DATE, PROVINCE, REGION) %>%
+        tally(CASES) %>%
+        ungroup()
+
+      hosp_data <- self$data$raw$hosp %>%
+        select(DATE, REGION, PROVINCE, NEW_IN) %>%
+        mutate(
+          DATE = ymd(DATE),
+          NEW_IN = as.numeric(NEW_IN)
+        ) %>%
+        replace_na(list(
+          REGION = "Unknown",
+          PROVINCE = "Unknown"
+        )) %>%
+        group_by(DATE, PROVINCE, REGION) %>%
+        tally(wt = NEW_IN) %>%
+        ungroup()
+
+      # Join the two datasets and rename columns
+      self$data$clean <- full_join(cases_data, hosp_data,
+                                   by = c("DATE",
+                                          "PROVINCE",
+                                          "REGION")) %>%
+        rename(date = DATE,
+               level_1_region = REGION,
+               level_2_region = PROVINCE,
+               cases_new = n.x,
+               hosp_new = n.y) %>%
+        left_join(self$codes_lookup[[2]],
```

Comment:
Do we want to include level 1 codes here/can we?

Reply:
Level 1 codes are included in the level 2 look-up table. The level 1 name is already in the source table, so we're just filling things out here, though (as noted elsewhere) both `level_1_region_code` and `level_2_region_code` are ISO 3166-2 codes.